mathics/core/expression.py

```diff
@@ -836,7 +836,14 @@ def get_sort_key(self, pattern_sort=False) -> tuple:
                     1,
                 )
             else:
-                return (1 if self.is_numeric() else 2, 3, head, self._elements, 1)
+                return (
+                    1 if self.is_numeric() else 2,
+                    3,
+                    head,
+                    len(self._elements),
```

Comment:
This new field ensures that the expressions with a smaller number of elements come first.